optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryProvider.java

```diff
@@ -0,0 +1,36 @@
+package org.optaplanner.core.impl.score.director;
+
+import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig;
+import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
+
+/**
+ * All {@link ScoreDirectorFactory} implementations must provide an implementation of this interface,
+ * as well as an entry in META-INF/services/org.optaplanner.core.impl.score.director.ScoreDirectorFactoryProvider file.
+ * This makes it available for discovery in {@link ScoreDirectorFactoryFactory} via {@link java.util.ServiceLoader}.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <Score_> the score type to go with the solution
+ */
+public interface ScoreDirectorFactoryProvider<Solution_, Score_ extends Score<Score_>> {
```

Comment:
Are we sure about this name?
Proposal A) ScoreDirectorFactoryProvider. Might be a bit close to "ConstraintProvider" and "ValueRangeProvider" with which I don't see the similarity.
Proposal B) ScoreDirectorFactoryBuilder. Doesn't really follow the "Builder" design pattern.
Proposal C) ScoreDirectorFactorySupplier. doesn't implement the Supplier function
Proposal D) ScoreDirectorFactoryProvisioner

I don't know. Your call. My preference order: D, A, B, C